src/main.rs

```diff
@@ -36,11 +51,29 @@ struct TypeSizeCmd {
 
 impl TypeSizeCmd {
     fn new(old_args: impl Iterator<Item = String>) -> Self {
-        let cargo_subcommand = "check";
+        let mut cargo_subcommand = "check";
         let args = vec![];
         let mut typesize_args: Vec<String> = vec![];
 
-        typesize_args.extend(old_args);
+        for arg in old_args {
+            match arg.as_str() {
+                "--build" => {
+                    cargo_subcommand = "build";
+                    continue;
+                }
+                "--check" => {
+                    cargo_subcommand = "check";
+                    continue;
+                }
+                "--test" => {
+                    cargo_subcommand = "test";
+                    continue;
```

Comment:
I noticed it was running the test cases when using `--test`, lets add `--no-run` to the args list to prevent that.

```suggestion
                    cargo_subcommand = "test";
                    args.push("--no-run".to_string());
                    continue;
```